scripts/build-types.mjs

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-console, no-process-exit, functional/no-expression-statement */
+import path from 'path';
+import * as url from 'url';
+
+import { exec } from './utils/exec.mjs';
+
+const apiExtractorPath = path.resolve(
+  url.fileURLToPath(new URL('.', import.meta.url)), // __dirname es module style
+  '../node_modules/.bin/api-extractor'
+);
+
+const args = process.argv.slice(2);
+
+/**
+ * ApiExtractor cli args.
+ */
+const apiExtractorArgs = ['run', '--local', ...args];
+
+// let apiExtractorArgs = ['run',  ...args];
+
+// When in CI `api-extractor` shouldn't apply the local flag, `api-extractor` will error if an api
+// change has occurred without being committed in an `api.md` file.
+// if (process.env.CI) {
+//   if (args.includes('--local')) {
+//     const index = args.indexOf('--local');
+//     apiExtractorArgs = [
+//       ...apiExtractorArgs.slice(0, index),
+//       ...apiExtractorArgs.slice(index + 1, apiExtractorArgs.length),
+//     ];
+//   }
+// } else if (!args.includes('--local')) {
+//   apiExtractorArgs = [...apiExtractorArgs, '--local'];
+// }
```

Comment:
Once we are ready for `api-extractor` to enforce the api in the `api.md` files, we just need to uncomment this and it will apply `--local` only if the `CI` env variable isn't truthy.